source/common/network/filter_manager_impl.h

```diff
@@ -11,35 +11,114 @@ namespace Envoy {
 namespace Network {
 
 /**
- * Interface used to obtain read/write buffers.
+ * Declaration of StreamBuffer backwards compatible with the former BufferSource::StreamBuffer.
  */
-class BufferSource {
+class StreamBufferSource {
 public:
-  virtual ~BufferSource() {}
+  virtual ~StreamBufferSource() {}
 
   struct StreamBuffer {
     Buffer::Instance& buffer;
     bool end_stream;
   };
+};
+
+/**
+ * Interface used to obtain read buffers.
+ */
+class ReadBufferSource : public virtual StreamBufferSource {
+public:
+  virtual ~ReadBufferSource() {}
 
   /**
    * Fetch the read buffer for the source.
    */
   virtual StreamBuffer getReadBuffer() PURE;
+};
+
+/**
+ * Interface used to obtain write buffers.
+ */
+class WriteBufferSource : public virtual StreamBufferSource {
+public:
+  virtual ~WriteBufferSource() {}
 
   /**
    * Fetch the write buffer for the source.
    */
   virtual StreamBuffer getWriteBuffer() PURE;
 };
 
+/**
+ * Interface used to obtain read/write buffers.
+ */
+class BufferSource : public ReadBufferSource, public WriteBufferSource {
+public:
+  virtual ~BufferSource() {}
+};
+
+/**
+ * Adapter that masquerades a given buffer instance as a ReadBufferSource.
+ */
+class FixedReadBufferSource : public ReadBufferSource {
+public:
+  FixedReadBufferSource(Buffer::Instance& data, bool end_stream)
+      : data_(data), end_stream_(end_stream) {}
+
+  ~FixedReadBufferSource() {}
```

Comment:
nit: del

Reply:
Sorry, I'm not following.

Is it possible to omit constructor without making data members public ?

Reply:
You don't need the empty destructor, it doesn't do anything.

Reply:
Ok

Reply:
Fixed